pkg/detectors/dockerhubv2/dockerhubv2.go

```diff
@@ -0,0 +1,179 @@
+package dockerhubv2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/golang-jwt/jwt/v4"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+func (s Scanner) Version() int { return 2 }
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.Versioner = (*Scanner)(nil)
+
+var (
+	// Can use email or username for login.
+	usernamePat = regexp.MustCompile(`(?im)(?:user|usr|-u|id)\S{0,40}?[:=\s]{1,3}[ '"=]?([a-zA-Z0-9]{4,40})\b`)
+	emailPat    = regexp.MustCompile(`(` + common.EmailPattern + `)`)
+
+	// Can use password or personal access token (PAT) for login, but this scanner will only check for PATs.
+	accessTokenPat = regexp.MustCompile(detectors.PrefixRegex([]string{"docker"}) + `\b([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12})\b`)
```

Comment:
Wouldn't this technically be v1? Is there a semantic meaning to versions, or does it not matter if an older format is "v2"? (Question for the TruffleSec team.)

Reply:
i did the same in my previous PR and the trufflesec team was not happy abt it and wanted me to make this one a v2. So i did.

Reply:
I've updated the PR to our current package structure which is

pkg/detectors/dockerhub/v1
pkg/detectors/dockerhub/v2